radio/src/targets/common/arm/stm32/timers_driver.cpp

```diff
@@ -32,10 +32,10 @@ void init2MhzTimer()
   TIMER_2MHz_TIMER->CR1 = TIM_CR1_CEN;
 }
 
-// Start TIMER at 1000Hz
+// Start TIMER at 10kHz
 void init1msTimer()
```

Comment:
The more appropriate name for the function would be `init100usTimer()`

Reply:
With presently max supported RF module rate of 1 kHz, I am sceptical if we need to increase computational load for the MCU to run the Flysky gimbal inter/extrapolation calculation 10 times faster. As the values are needed max. at 1 kHz.

Reply:
@rotorman Is there any better option than using timer for UART data processing?

Reply:
Even though using 10kHz timer, it seems the timing still quite raw.  However, this PR is used for concept verification.  I think best resampling is in the gimbal firmware.  So it will be perfect if Flysky can update the firmware to provide 2 modes: 1. sync 250Hz, 2. Resampled output at any freq.

Reply:
As long as Flysky digital hall gimbals are not yet able to do polled mode and just stream out data at their own pace, why not use the mixer task to trigger the interpolation/extrapolation calculation, as then the values are created as they are needed? Thus no async timer controlled triggering for the calculation is needed IMO.

Reply:
> As long as Flysky digital hall gimbals are not yet able to do polled mode and just stream out data at their own pace, why not use the mixer task to trigger the interpolation/extrapolation calculation, as then the values are created as they are needed? Thus no async timer controlled triggering for the calculation is needed IMO.
> 
> The gimbal data reception task (UART RX ISR) could save couple of last incoming values in a circular buffer, here again then in event triggered fashion.

This is the current implementation, the timer task is used to save the incoming values with proper timings.  If the circular buffer is not consumed on time, the timing will be wrong.  Well the best place for timing is in the Gimbal firmware anyway.  I will see if I can persuade Flysky to work this out.

Reply:
Hi @richardclli , any luck talking to Flysky to update gimbals?